index.js

```diff
@@ -31,9 +31,18 @@ const mapObject = (object, mapper, options, isSeen = new WeakMap()) => {
 	}
 
 	for (const [key, value] of Object.entries(object)) {
-		let [newKey, newValue] = mapper(key, value, object);
+		let [newKey, newValue, mapperOptions = {shouldRecurse: true}] = mapper(key, value, object);
 
-		if (options.deep && isObjectCustom(newValue)) {
+		if (typeof mapperOptions.shouldRecurse !== 'boolean') {
```

Comment:
edge case where someone passes an empty object for `mapperOptions`, and shouldRecurse doesn't get a default. Do we care about this?

Reply:
updated the readme, this is now explicit

Reply:
Passing an empty object should be allowed.